rules/Carbon/NodeFactory/CarbonCallFactory.php

```diff
@@ -12,116 +11,64 @@
 
     /**
      * @var string
-     * @see https://regex101.com/r/WILFvX/1
+     * @see https://regex101.com/r/IhxHTO/1
      */
-    private const MINUS_MINUTE_COUNT_REGEX = '#-(\s+)?(?<count>\d+)(\s+)?(minute|minutes)#';
+    private const STATIC_DATE_REGEX = '#now|yesterday|today|tomorrow#';
 
     /**
-     * @var string
-     * @see https://regex101.com/r/FwCUup/1
+     * @return \PhpParser\Node\Expr\MethodCall|\PhpParser\Node\Expr\StaticCall
      */
-    private const MINUS_SECOND_COUNT_REGEX = '#-(\s+)?(?<count>\d+)(\s+)?(second|seconds)#';
-
-    /**
-     * @var array<self::*_REGEX, string>
-     */
-    private const REGEX_TO_METHOD_NAME_MAP = [
-        self::PLUS_DAY_COUNT_REGEX => 'addDays',
-        self::MINUS_DAY_COUNT_REGEX => 'subDays',
-        self::PLUS_MONTH_COUNT_REGEX => 'addMonths',
-        self::MINUS_MONTH_COUNT_REGEX => 'subMonths',
-        self::PLUS_HOUR_COUNT_REGEX => 'addHours',
-        self::MINUS_HOUR_COUNT_REGEX => 'subHours',
-        self::PLUS_MINUTE_COUNT_REGEX => 'addMinutes',
-        self::MINUS_MINUTE_COUNT_REGEX => 'subMinutes',
-        self::PLUS_SECOND_COUNT_REGEX => 'addSeconds',
-        self::MINUS_SECOND_COUNT_REGEX => 'subSeconds',
-    ];
-
-    public function createFromDateTimeString(
-        FullyQualified $carbonFullyQualified,
-        String_ $string
-    ): MethodCall|StaticCall {
+    public function createFromDateTimeString(FullyQualified $carbonFullyQualified, String_ $string)
+    {
         $dateTimeValue = $string->value;
-        if ($dateTimeValue === 'now') {
-            return new StaticCall($carbonFullyQualified, new Identifier('now'));
+        if (in_array($dateTimeValue, ['now', 'today', 'yesterday', 'tomorrow'])) {
+            return new StaticCall($carbonFullyQualified, new Identifier($dateTimeValue));
         }
 
-        if ($dateTimeValue === 'today') {
-            return new StaticCall($carbonFullyQualified, new Identifier('today'));
-        }
-
-        $hasToday = Strings::match($dateTimeValue, '#today#');
-        if ($hasToday !== null) {
-            $carbonCall = new StaticCall($carbonFullyQualified, new Identifier('today'));
-        } else {
-            $carbonCall = new StaticCall($carbonFullyQualified, new Identifier('now'));
-        }
+        $startDate = Strings::match($dateTimeValue, self::STATIC_DATE_REGEX)[0] ?? 'now';
+        $carbonCall = new StaticCall($carbonFullyQualified, new Identifier($startDate));
 
-        foreach (self::REGEX_TO_METHOD_NAME_MAP as $regex => $methodName) {
-            $match = Strings::match($dateTimeValue, $regex);
-            if ($match === null) {
-                continue;
-            }
+        $match = Strings::match($dateTimeValue, self::PLUS_MINUS_COUNT_REGEX);
 
-            $countLNumber = new LNumber((int) $match['count']);
+        if ($match === null) {
+            return new StaticCall($carbonFullyQualified, new Identifier('parse'), [new Arg($string)]);
```

Comment:
can this just return `null`, then verify if return null, then no change, that's ensure the transformation is always safe.

Reply:
@samsonasik I don't think I really understand what you mean here? 
You don't want to use the parse function and just keep the \DateTime if it doesn't match?

Reply:
yes, if it doesn't match, just skip the transformation

Reply:
But why? We want to migrate all instances of \DateTime to carbon.
If we have things like new \DateTime('2024-01-25') we want this as Carbon date. Fallback to parse method will make sure this always work this is a drop in replacement of the datetime constructor.

So why leave it without transformation?

Reply:
I see, let's @TomasVotruba decide ;)